src/analysis/processing/qgsalgorithmaddincrementalfield.cpp

```diff
@@ -117,6 +118,12 @@ bool QgsAddIncrementalFieldAlgorithm::prepareAlgorithm( const QVariantMap &param
   mSortAscending = parameterAsBoolean( parameters, QStringLiteral( "SORT_ASCENDING" ), context );
   mSortNullsFirst = parameterAsBoolean( parameters, QStringLiteral( "SORT_NULLS_FIRST" ), context );
 
+  if ( source->fields().lookupField( mFieldName ) >= 0 )
+  {
+    feedback->reportError( QObject::tr( "Field with the same name already exists" ) );
```

Comment:
we should raise the exception here instead, given that it's a fatal error which prevents the algorithm running:

```suggestion
    throw QgsProcessingException( QObject::tr( "Field with the same name already exists" ) );
```